probe-rs/src/architecture/arm/memory/romtable.rs

```diff
@@ -81,7 +81,7 @@ impl<'probe, 'memory, 'reader> Iterator for RomTableIterator<'probe, 'memory, 'r
             .memory
             .read_32(component_address, &mut entry_data)
         {
-            return Some(Err(RomTableError::Memory(e)));
+            return Some(Err(RomTableError::Memory(Box::new(e))));
```

Comment:
How about a `memory()` function that boxes the error and returns the constructed `RomTableError::Memory` variant? :)